codec-http/src/main/java/io/netty/handler/codec/http/HttpServerCodec.java

```diff
@@ -96,16 +97,23 @@ public HttpServerRequestDecoder(int maxInitialLineLength, int maxHeaderSize, int
         }
 
         @Override
-        protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> out) throws Exception {
-            int oldSize = out.size();
-            super.decode(ctx, buffer, out);
-            int size = out.size();
-            for (int i = oldSize; i < size; i++) {
-                Object obj = out.get(i);
-                if (obj instanceof HttpRequest) {
-                    queue.add(((HttpRequest) obj).method());
+        protected void decode(final ChannelHandlerContext ctx, ByteBuf buffer) throws Exception {
+            super.decode(context, buffer);
+        }
+
+        @Override
+        protected void handlerAdded0(final ChannelHandlerContext ctx) {
+            context = new DelegatingChannelHandlerContext(ctx) {
+
+                @Override
+                public ChannelHandlerContext fireChannelRead(Object msg) {
+                    if (msg instanceof HttpRequest) {
+                        queue.add(((HttpRequest) msg).method());
```

Comment:
I haven't played with HttpServerCodec.java but this doesn't feel the same as the previous implementation where it delegated so the next handler, then pushed the read objects into the queue..?

Reply:
@johnou its actually doing exactly the same as the object that were in the queue were fired though the next handler.